testng-core/src/main/java/org/testng/internal/invokers/TestNgMethodUtils.java

```diff
@@ -69,23 +71,31 @@ private static boolean containsConfigurationMethod(
   }
 
   static ITestNGMethod[] filterBeforeTestMethods(
-      ITestClass testClass, BiPredicate<ITestNGMethod, IClass> predicate) {
-    return filterMethods(testClass, testClass.getBeforeTestMethods(), predicate);
+      Object instance, ITestClass testClass, BiPredicate<ITestNGMethod, IClass> predicate) {
+    return filterMethods(instance, testClass, testClass.getBeforeTestMethods(), predicate);
   }
 
   static ITestNGMethod[] filterAfterTestMethods(
-      ITestClass testClass, BiPredicate<ITestNGMethod, IClass> predicate) {
-    return filterMethods(testClass, testClass.getAfterTestMethods(), predicate);
+      Object instance, ITestClass testClass, BiPredicate<ITestNGMethod, IClass> predicate) {
+    return filterMethods(instance, testClass, testClass.getAfterTestMethods(), predicate);
   }
 
   /** @return Only the ITestNGMethods applicable for this testClass */
   static ITestNGMethod[] filterMethods(
-      IClass testClass, ITestNGMethod[] methods, BiPredicate<ITestNGMethod, IClass> predicate) {
+      Object instance,
+      IClass testClass,
+      ITestNGMethod[] methods,
+      BiPredicate<ITestNGMethod, IClass> predicate) {
     List<ITestNGMethod> vResult = Lists.newArrayList();
+    Predicate<ITestNGMethod> sameInstance =
```

Comment:
It sounds like over-engineering here. Why not using a static method instead?

Reply:
No specific reason. the static method is also going to do the same thing as the predicate. Have flipped this to be a static method instead